httpx/models.py

```diff
@@ -316,8 +316,22 @@ def get(self, key: typing.Any, default: typing.Any = None) -> typing.Any:
             return self._dict[key]
         return default
 
+    def update(self, params: QueryParamTypes = None) -> None:  # type: ignore
+        if params:
+            params = QueryParams(params)
+        else:
+            params = QueryParams()
+        for param in params:
+            self[param] = params[param]
+
     def __getitem__(self, key: typing.Any) -> str:
```

Comment:
I don't think params are case insensitive per Requests implementation, it looks like they're just using a normal dictionary? I wasn't able to replicate the behavior being tested below:

```python
import requests

sess = requests.Session()
sess.params["a"] = "b"
assert session.params["A"] == "b"  # This fails
sess.params  # {"a": "b"}
```

Reply:
Ah great, that makes it easier then!